certbot/compat/os.py

```diff
@@ -29,3 +29,14 @@
 
 # Clean all remaining importables that are not from the core os module.
 del ourselves, std_os, std_sys
+
+
+# Because of the blocking strategy on file handlers on Windows, rename to not behave as expected
+# with POSIX systems: an exception will be raised if dst already exists. Hopefully there is
+# os.replace on Windows for Python 3, that will do the same than on POSIX. Hopefully also, only
+# Python 3 is supported for Certbot. So we can rely on os.rename on Linux, and os.replace
+# on Windows.
+def rename(*unused_args, **unused_kwargs):  # pylint: disable=function-redefined
+    """Method os.rename() is forbidden"""
+    raise RuntimeError('Usage of os.rename() is forbidden. '  # pragma: no cover
```

Comment:
Same comment as I put at https://github.com/certbot/certbot/pull/6964#discussion_r283070863.

Reply:
Done. As a side not, if part3a is merged first, I will update from this PR the test that trigger exceptions from `os` module. Otherwise, I will update it from part3a.